app/src/main/java/cm/aptoide/pt/promotions/PromotionsPresenter.java

```diff
@@ -151,11 +163,24 @@ private void getPromotionApps() {
     view.getLifecycleEvent()
         .filter(lifecycleEvent -> lifecycleEvent.equals(View.LifecycleEvent.CREATE))
         .doOnNext(__ -> view.showLoading())
-        .flatMap(__ -> loadPromotionApps())
+        .flatMap(__ -> promotionsManager.getPromotionsModel(promotionId))
+        .doOnNext(__ -> promotionsAnalytics.sendOpenPromotionsFragmentEvent())
+        .observeOn(viewScheduler)
+        .doOnNext(promotionsModel -> view.showAppCoinsAmount((promotionsModel.getTotalAppcValue())))
```

Comment:
Not problematic but we are showing the appcoins value, without needing it (in case there are no promotions). Maybe this should be moved bellow to the else case where it is not empty?

Reply:
fixed